phpBB/phpbb/db/migration/schema_generator.php

```diff
@@ -84,171 +102,245 @@ public function get_schema()
 		}
 
 		$migrations = $this->class_names;
+		$filter = function($class_name) {
+			return !migrator::is_migration($class_name);
+		};
+
+		$edges = function($class_name) {
+			return $class_name::depends_on();
+		};
+
+		$apply_for_each = function($class_name) {
+			$this->apply_migration_to_schema($class_name);
+		};
+
+		try
+		{
+			require_once __DIR__ . '/../../../vendor/chita/topological_sort/src/TopologicalSort.php';
```

Comment:
Not a super fan of this. Maybe use phpBB's root path instead? At least then it wouldn't be hardcoded ;)

Reply:
Or I can downgrade to 1.x of my lib, which places this into the autoloader. I don't really care either way if you have a preference. It is just a shame that PHP cannot autoload functions.